relecov_tools/read_bioinfo_metadata.py

```diff
@@ -820,28 +856,26 @@ def extract_file(self, file, dest_folder, sample_name=None, path_key=None):
         """
         dest_folder = os.path.join(dest_folder, "analysis_results")
         os.makedirs(dest_folder, exist_ok=True)
-        out_filepath = os.path.join(dest_folder, os.path.basename(file))
-        if os.path.isfile(out_filepath):
-            return True
-        if file == "Not Provided [GENEPIO:0001668]":
-            self.update_all_logs(
-                self.extract_file.__name__,
-                "warning",
-                f"File for {path_key} not provided in sample {sample_name}",
-            )
-            return False
-        try:
-            if re.search(r".*pangolin\.csv$", os.path.basename(file), re.IGNORECASE):
-                df = pd.read_csv(file)
-                df["lineage_analysis_date"] = datetime.now().strftime("%Y%m%d")
-                df.to_csv(out_filepath, index=False)
-            else:
-                shutil.copy(file, out_filepath)
-        except (IOError, PermissionError) as e:
-            self.update_all_logs(
-                self.extract_file.__name__, "warning", f"Could not extract {file}: {e}"
-            )
-            return False
+        for filepath in file:
+            out_filepath = os.path.join(dest_folder, os.path.basename(filepath))
+            if os.path.isfile(out_filepath):
+                return True
+            if filepath == "Not Provided [GENEPIO:0001668]":
+                self.update_all_logs(
+                    self.extract_file.__name__,
+                    "warning",
+                    f"File for {path_key} not provided in sample {sample_name}",
+                )
+                return False
```

Comment:
Here if there are multiple filepaths for that sample you cancel the operation at the first file with the two `return` statements, I think its more convenient to extract all the files that are correct, while keeping the log of those that failed.
```suggestion
            if os.path.isfile(out_filepath):
                self.log.debug(f"{out_filepath} already exists, not extracted")
                continue
            if filepath == "Not Provided [GENEPIO:0001668]":
                self.update_all_logs(
                    self.extract_file.__name__,
                    "warning",
                    f"File for {path_key} not provided in sample {sample_name}",
                )
                continue
```